File: internal/email/luckmail.go
package email

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
	"strconv"
	"strings"
	"time"
)

// LuckMailConfig LuckMail 配置
type LuckMailConfig struct {
	Name        string `json:"name"`        // 配置名称
	Token       string `json:"token"`       // 接口秘钥
	ProjectCode string `json:"projectCode"` // 项目代码
	EmailType   string `json:"emailType"`   // 邮箱类型 (ms_graph/ms_imap/google_variant/self_built, 可选)
	Domain      string `json:"domain"`      // 指定域名 (可选)
	BaseURL     string `json:"baseURL"`     // API 地址 (可选，默认 https://mails.luckyous.com)
}

// LuckMailClient LuckMail API 客户端
type LuckMailClient struct {
	config  LuckMailConfig
	client  *http.Client
	baseURL string
}

const defaultLuckMailBaseURL = "https://mails.luckyous.com"

// NewLuckMailClient 创建 LuckMail 客户端
func NewLuckMailClient(config LuckMailConfig) *LuckMailClient {
	baseURL := config.BaseURL
	if baseURL == "" {
		baseURL = defaultLuckMailBaseURL
	}
	// 移除末尾的斜杠
	baseURL = strings.TrimSuffix(baseURL, "/")

	return &LuckMailClient{
		config:  config,
		client:  &http.Client{Timeout: 15 * time.Second},
		baseURL: baseURL,
	}
}

// sign 生成 HMAC-SHA256 签名
func (c *LuckMailClient) sign(method, path, timestamp, body string) string {
	message := method + path + timestamp + body
	mac := hmac.New(sha256.New, []byte(c.config.Token))
	mac.Write([]byte(message))
	return hex.EncodeToString(mac.Sum(nil))
}

// request 发送带签名的 HTTP 请求
func (c *LuckMailClient) request(method, path string, body string) ([]byte, error) {
	url := c.baseURL + path
	timestamp := fmt.Sprintf("%d", time.Now().Unix())
	signature := c.sign(method, path, timestamp, body)

	var bodyReader io.Reader
	if body != "" {
		bodyReader = strings.NewReader(body)
	}

	req, err := http.NewRequest(method, url, bodyReader)
	if err != nil {
		return nil, err
	}

	req.Header.Set("X-API-Key", c.config.Token)
	req.Header.Set("X-Timestamp", timestamp)
	req.Header.Set("X-Signature", signature)
	if body != "" {
		req.Header.Set("Content-Type", "application/json")
	}

	resp, err := c.client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("请求失败: %w", err)
	}
	defer resp.Body.Close()

	respBody, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, fmt.Errorf("读取响应失败: %w", err)
	}

	return respBody, nil
}

// LuckMailResponse 统一响应格式
type LuckMailResponse struct {
	Code    int             `json:"code"`
	Message string          `json:"message"`
	Data    json.RawMessage `json:"data"`
}

// LuckMailOrderData 创建订单响应数据
type LuckMailOrderData struct {
	OrderNo        string `json:"order_no"`
	EmailAddress   string `json:"email_address"`
	Project        string `json:"project"`
	Price          string `json:"price"`          // API 返回字符串
	TimeoutSeconds int    `json:"timeout_seconds"`
	ExpiredAt      string `json:"expired_at"`
}

// LuckMailCodeData 轮询验证码响应数据
type LuckMailCodeData struct {
	OrderNo          string `json:"order_no"`
	Status           string `json:"status"` // pending/success/timeout/cancelled
	VerificationCode string `json:"verification_code"`
	MailFrom         string `json:"mail_from"`
	MailSubject      string `json:"mail_subject"`
}

// LuckMailProjectItem 项目列表项
type LuckMailProjectItem struct {
	Code string `json:"code"`
	Name string `json:"name"`
}

// LuckMailDomainItem 域名列表项
type LuckMailDomainItem struct {
	Domain    string `json:"domain"`
	EmailType string `json:"email_type"`
	Count     int    `json:"count,omitempty"` // 可用数量（如果API返回）
}

var defaultLuckMailDomains = []LuckMailDomainItem{
	// Microsoft Graph 域名
	{Domain: "outlook.com", EmailType: "ms_graph", Count: 361802},
	{Domain: "hotmail.com", EmailType: "ms_graph", Count: 133313},
	{Domain: "outlook.de", EmailType: "ms_graph", Count: 3392},
	{Domain: "outlook.es", EmailType: "ms_graph"},
	// Microsoft IMAP 域名
	{Domain: "hotmail.com", EmailType: "ms_imap", Count: 103083},
	{Domain: "outlook.com", EmailType: "ms_imap", Count: 63247},
	{Domain: "outlook.cl", EmailType: "ms_imap", Count: 37914},
	{Domain: "outlook.my", EmailType: "ms_imap", Count: 25828},
	{Domain: "outlook.fr", EmailType: "ms_imap"},
	// 自建邮箱域名
	{Domain: "mail.agentsforge.org", EmailType: "self_built"},
	{Domain: "a1.agentsforge.org", EmailType: "self_built"},
	{Domain: "mails.chatoai.pro", EmailType: "self_built"},
	{Domain: "ak.chatoai.pro", EmailType: "self_built"},
	{Domain: "mails.qubitsforge.org", EmailType: "self_built"},
	{Domain: "ewsbugwih.us.ci", EmailType: "self_built"},
	{Domain: "mdw.ggband.dev", EmailType: "self_built"},
	{Domain: "phw.ggband.dev", EmailType: "self_built"},
	{Domain: "dmw.ggband.dev", EmailType: "self_built"},
	{Domain: "eru.zhanggui.me", EmailType: "self_built"},
	{Domain: "barbararamey.us.ci", EmailType: "self_built"},
	{Domain: "josephdutcher.eu.cc", EmailType: "self_built"},
	{Domain: "jimmykelley.eu.cc", EmailType: "self_built"},
	{Domain: "dorseybarber.eu.cc", EmailType: "self_built"},
	{Domain: "ryanrene.dpdns.org", EmailType: "self_built"},
	{Domain: "ryanrene.qzz.io", EmailType: "self_built"},
	{Domain: "mail2.qq120.ip-ddns.com", EmailType: "self_built"},
	{Domain: "mail3.qq120.ip-ddns.com", EmailType: "self_built"},
	{Domain: "mail4.qq120.ip-ddns.com", EmailType: "self_built"},
	{Domain: "mail1.tc712.cloud-ip.cc", EmailType: "self_built"},
	{Domain: "mail2.tc712.cloud-ip.cc", EmailType: "self_built"},
	{Domain: "mail3.tc712.cloud-ip.cc", EmailType: "self_built"},
	{Domain: "mail1.tc713.abrdns.com", EmailType: "self_built"},
	{Domain: "mail2.tc713.abrdns.com", EmailType: "self_built"},
	{Domain: "mail3.tc713.abrdns.com", EmailType: "self_built"},
	{Domain: "caijiuduolian.bbroot.com", EmailType: "self_built"},
}

// GetDefaultLuckMailDomains 获取内置域名列表
func GetDefaultLuckMailDomains(emailType string) []LuckMailDomainItem {
	if emailType == "" {
		return append([]LuckMailDomainItem(nil), defaultLuckMailDomains...)
	}

	domains := make([]LuckMailDomainItem, 0)
	for _, domain := range defaultLuckMailDomains {
		if domain.EmailType == emailType {
			domains = append(domains, domain)
		}
	}
	return domains
}

// GetBalance 查询余额
func (c *LuckMailClient) GetBalance() (float64, error) {
	respBody, err := c.request("GET", "/api/v1/openapi/balance", "")
	if err != nil {
		return 0, err
	}

	var resp LuckMailResponse
	if err := json.Unmarshal(respBody, &resp); err != nil {
		return 0, fmt.Errorf("解析响应失败: %w", err)
	}
	if resp.Code != 0 {
		return 0, fmt.Errorf("API 错误 (%d): %s", resp.Code, resp.Message)
	}

	var data map[string]interface{}
	json.Unmarshal(resp.Data, &data)

	// 尝试多种方式解析余额
	var balance float64
	if balanceVal, ok := data["balance"]; ok {
		switch v := balanceVal.(type) {
		case float64:
			balance = v
		case string:
			// 字符串转 float64
			if f, err := strconv.ParseFloat(v, 64); err == nil {
				balance = f
			}
		case json.Number:
			if f, err := v.Float64(); err == nil {
				balance = f
			}
		}
	}

	return balance, nil
}

// GetProjects 获取可用项目列表
func (c *LuckMailClient) GetProjects() ([]LuckMailProjectItem, error) {
	respBody, err := c.request("GET", "/api/v1/openapi/projects", "")
	if err != nil {
		return nil, err
	}

	var resp LuckMailResponse
	if err := json.Unmarshal(respBody, &resp); err != nil {
		return nil, fmt.Errorf("解析响应失败: %w", err)
	}
	if resp.Code != 0 {
		return nil, fmt.Errorf("API 错误 (%d): %s", resp.Code, resp.Message)
	}

	var data struct {
		List []LuckMailProjectItem `json:"list"`
	}
	json.Unmarshal(resp.Data, &data)
	return data.List, nil
}

// GetDomains 获取可用域名列表
func (c *LuckMailClient) GetDomains() ([]LuckMailDomainItem, error) {
	respBody, err := c.request("GET", "/api/v1/openapi/domains", "")
	if err != nil {
		return nil, err
	}

	var resp LuckMailResponse
	if err := json.Unmarshal(respBody, &resp); err != nil {
		return nil, fmt.Errorf("解析响应失败: %w", err)
	}
	if resp.Code != 0 {
		return nil, fmt.Errorf("API 错误 (%d): %s", resp.Code, resp.Message)
	}

	var data struct {
		List []LuckMailDomainItem `json:"list"`
	}
	json.Unmarshal(resp.Data, &data)
	return data.List, nil
}

// CreateOrder 创建接码订单
func (c *LuckMailClient) CreateOrder() (*LuckMailOrderData, error) {
	reqData := map[string]interface{}{
		"project_code": c.config.ProjectCode,
	}
	if c.config.EmailType != "" {
		reqData["email_type"] = c.config.EmailType
	}
	if c.config.Domain != "" {
		reqData["domain"] = c.config.Domain
	}

	reqJSON, _ := json.Marshal(reqData)
	respBody, err := c.request("POST", "/api/v1/openapi/order/create", string(reqJSON))
	if err != nil {
		return nil, err
	}

	var resp LuckMailResponse
	if err := json.Unmarshal(respBody, &resp); err != nil {
		return nil, fmt.Errorf("解析响应失败: %w", err)
	}
	if resp.Code != 0 {
		return nil, fmt.Errorf("创建订单失败 (%d): %s", resp.Code, resp.Message)
	}

	var order LuckMailOrderData
	if err := json.Unmarshal(resp.Data, &order); err != nil {
		return nil, fmt.Errorf("解析订单数据失败: %w", err)
	}
	return &order, nil
}

// PollCode 轮询验证码
func (c *LuckMailClient) PollCode(orderNo string) (*LuckMailCodeData, error) {
	path := "/api/v1/openapi/order/" + orderNo + "/code"
	respBody, err := c.request("GET", path, "")
	if err != nil {
		return nil, err
	}

	var resp LuckMailResponse
	if err := json.Unmarshal(respBody, &resp); err != nil {
		return nil, fmt.Errorf("解析响应失败: %w", err)
	}
	if resp.Code != 0 {
		return nil, fmt.Errorf("查询验证码失败 (%d): %s", resp.Code, resp.Message)
	}

	var data LuckMailCodeData
	if err := json.Unmarshal(resp.Data, &data); err != nil {
		return nil, fmt.Errorf("解析验证码数据失败: %w", err)
	}
	return &data, nil
}

// CancelOrder 取消订单
func (c *LuckMailClient) CancelOrder(orderNo string) error {
	path := "/api/v1/openapi/order/" + orderNo + "/cancel"
	respBody, err := c.request("POST", path, "")
	if err != nil {
		return err
	}

	var resp LuckMailResponse
	json.Unmarshal(respBody, &resp)
	if resp.Code != 0 {
		return fmt.Errorf("取消订单失败 (%d): %s", resp.Code, resp.Message)
	}
	return nil
}

// TestConnection 测试连接（通过查询余额验证）
func (c *LuckMailClient) TestConnection() (float64, error) {
	return c.GetBalance()
}

// LuckMailProvider 实现邮箱提供商
type LuckMailProvider struct {
	client  *LuckMailClient
	orderNo string
	address string
}

// NewLuckMailProvider 创建 LuckMail 提供商（创建订单获取邮箱）
func NewLuckMailProvider(config LuckMailConfig) (*LuckMailProvider, error) {
	client := NewLuckMailClient(config)

	// 如果指定了域名，使用内置域名列表进行验证
	if config.Domain != "" {
		domains := GetDefaultLuckMailDomains(config.EmailType)

		if len(domains) > 0 {
			// 验证域名是否在可用列表中
			domainValid := false
			var availableDomains []string

			// 如果指定了 EmailType，只检查该类型的域名
			if config.EmailType != "" {
				for _, d := range domains {
					if d.EmailType == config.EmailType {
						availableDomains = append(availableDomains, d.Domain)
						if d.Domain == config.Domain {
							domainValid = true
						}
					}
				}
			} else {
				// 未指定类型，检查所有域名
				for _, d := range domains {
					availableDomains = append(availableDomains, d.Domain)
					if d.Domain == config.Domain {
						domainValid = true
					}
				}
			}

			if !domainValid {
				// 域名不可用，尝试使用第一个可用域名
				if len(availableDomains) > 0 {
					oldDomain := config.Domain
					config.Domain = availableDomains[0]
					log.Printf("[LuckMail] 域名 %s 不在可用列表中，自动切换到 %s", oldDomain, config.Domain)
				} else {
					return nil, fmt.Errorf("域名 %s 不在可用列表中，且没有其他可用域名", config.Domain)
				}
			}
		}
	}

	order, err := client.CreateOrder()
	if err != nil {
		return nil, fmt.Errorf("创建接码订单失败: %w", err)
	}

	log.Printf("[LuckMail] 订单创建成功: %s, 邮箱: %s, 超时: %ds",
		order.OrderNo, order.EmailAddress, order.TimeoutSeconds)

	return &LuckMailProvider{
		client:  client,
		orderNo: order.OrderNo,
		address: order.EmailAddress,
	}, nil
}

// GetAddress 返回邮箱地址
func (p *LuckMailProvider) GetAddress() string {
	return p.address
}

// WaitForCode 轮询等待验证码
func (p *LuckMailProvider) WaitForCode(timeout, interval int) (string, error) {
	maxRetries := timeout / interval
	log.Printf("[LuckMail] 开始等待验证码, 订单: %s", p.orderNo)

	for attempt := 1; attempt <= maxRetries; attempt++ {
		data, err := p.client.PollCode(p.orderNo)
		if err != nil {
			if attempt%5 == 0 {
				log.Printf("[LuckMail] 查询失败: %v, 重试中...", err)
			}
			time.Sleep(time.Duration(interval) * time.Second)
			continue
		}

		switch data.Status {
		case "success":
			log.Printf("[LuckMail] 验证码获取成功: %s", data.VerificationCode)
			return data.VerificationCode, nil
		case "timeout":
			return "", fmt.Errorf("LuckMail 订单超时")
		case "cancelled":
			return "", fmt.Errorf("LuckMail 订单已取消")
		case "pending":
			if attempt%5 == 0 {
				log.Printf("[LuckMail] [%d/%d] 等待验证码中...", attempt, maxRetries)
			}
		}

		time.Sleep(time.Duration(interval) * time.Second)
	}

	return "", fmt.Errorf("等待验证码超时 (%ds)", timeout)
}

// Cancel 取消订单（任务取消时调用）
func (p *LuckMailProvider) Cancel() {
	if p.orderNo != "" {
		if err := p.client.CancelOrder(p.orderNo); err != nil {
			log.Printf("[LuckMail] 取消订单失败: %v", err)
		} else {
			log.Printf("[LuckMail] 订单已取消: %s", p.orderNo)
		}
	}
}

File: frontend/js/luckmail.js
// ===== LuckMail 配置管理 =====

console.log('[LuckMail] luckmail.js 开始加载');

var luckmailConfigs = [];
var luckmailConfigStatus = {};
var luckmailDomains = []; // 缓存可用域名列表

var defaultLuckMailDomains = [
  // Microsoft Graph 域名
  { domain: 'outlook.com', email_type: 'ms_graph', count: 361802 },
  { domain: 'hotmail.com', email_type: 'ms_graph', count: 133313 },
  { domain: 'outlook.de', email_type: 'ms_graph', count: 3392 },
  { domain: 'outlook.es', email_type: 'ms_graph' },
  // Microsoft IMAP 域名
  { domain: 'hotmail.com', email_type: 'ms_imap', count: 103083 },
  { domain: 'outlook.com', email_type: 'ms_imap', count: 63247 },
  { domain: 'outlook.cl', email_type: 'ms_imap', count: 37914 },
  { domain: 'outlook.my', email_type: 'ms_imap', count: 25828 },
  { domain: 'outlook.fr', email_type: 'ms_imap' },
  // 自建邮箱域名
  { domain: 'mail.agentsforge.org', email_type: 'self_built' },
  { domain: 'a1.agentsforge.org', email_type: 'self_built' },
  { domain: 'mails.chatoai.pro', email_type: 'self_built' },
  { domain: 'ak.chatoai.pro', email_type: 'self_built' },
  { domain: 'mails.qubitsforge.org', email_type: 'self_built' },
  { domain: 'ewsbugwih.us.ci', email_type: 'self_built' },
  { domain: 'mdw.ggband.dev', email_type: 'self_built' },
  { domain: 'phw.ggband.dev', email_type: 'self_built' },
  { domain: 'dmw.ggband.dev', email_type: 'self_built' },
  { domain: 'eru.zhanggui.me', email_type: 'self_built' },
  { domain: 'barbararamey.us.ci', email_type: 'self_built' },
  { domain: 'josephdutcher.eu.cc', email_type: 'self_built' },
  { domain: 'jimmykelley.eu.cc', email_type: 'self_built' },
  { domain: 'dorseybarber.eu.cc', email_type: 'self_built' },
  { domain: 'ryanrene.dpdns.org', email_type: 'self_built' },
  { domain: 'ryanrene.qzz.io', email_type: 'self_built' },
  { domain: 'mail2.qq120.ip-ddns.com', email_type: 'self_built' },
  { domain: 'mail3.qq120.ip-ddns.com', email_type: 'self_built' },
  { domain: 'mail4.qq120.ip-ddns.com', email_type: 'self_built' },
  { domain: 'mail1.tc712.cloud-ip.cc', email_type: 'self_built' },
  { domain: 'mail2.tc712.cloud-ip.cc', email_type: 'self_built' },
  { domain: 'mail3.tc712.cloud-ip.cc', email_type: 'self_built' },
  { domain: 'mail1.tc713.abrdns.com', email_type: 'self_built' },
  { domain: 'mail2.tc713.abrdns.com', email_type: 'self_built' },
  { domain: 'mail3.tc713.abrdns.com', email_type: 'self_built' },
  { domain: 'caijiuduolian.bbroot.com', email_type: 'self_built' }
];

console.log('[LuckMail] 全局变量已初始化');

// 加载 LuckMail 配置
async function loadLuckMailConfigs() {
  console.log('[LuckMail] loadLuckMailConfigs 被调用');
  try {
    luckmailConfigs = await window.go.main.App.GetLuckMailConfigs() || [];
    renderLuckMailInlineList();
    updateLuckMailSummary();
  } catch (e) {
    console.error('加载 LuckMail 配置失败:', e);
  }
}

// 更新摘要
function updateLuckMailSummary() {
  var el = document.getElementById('settings-luckmail-summary');
  if (el) {
    el.textContent = luckmailConfigs.length > 0 ? luckmailConfigs.length + ' 个配置' : '未配置';
  }
}

// 渲染配置列表
function renderLuckMailInlineList() {
  var listDiv = document.getElementById('luckmail-inline-list');
  if (!listDiv) return;

  if (luckmailConfigs.length === 0) {
    listDiv.innerHTML = '<div style="text-align:center;color:var(--text-muted);font-size:12px;padding:16px;">暂无配置</div>';
    return;
  }

  try {
    var saved = localStorage.getItem('luckmail-config-status');
    if (saved) luckmailConfigStatus = JSON.parse(saved);
  } catch (e) {}

  var html = '';
  luckmailConfigs.forEach(function(cfg, idx) {
    var status = luckmailConfigStatus[cfg.name];
    var statusHtml = '';
    if (status && status.tested) {
      if (status.success) {
        statusHtml = '<span style="color:var(--success);font-size:11px;">余额: ¥' + (status.balance || 0).toFixed(2) + '</span>';
      } else {
        statusHtml = '<span style="color:var(--danger);font-size:11px;">连接失败</span>';
      }
    }

    html += '<div class="moemail-config-item">' +
      '<div class="moemail-config-info">' +
        '<div class="moemail-config-name">' + escapeHtml(cfg.name) + '</div>' +
        '<div class="moemail-config-url">' + escapeHtml(cfg.projectCode) + (cfg.emailType ? ' / ' + escapeHtml(cfg.emailType) : '') + '</div>' +
      '</div>' +
      '<div class="moemail-config-actions">' +
        statusHtml +
        '<button onclick="testLuckMailConfig(' + idx + ')" class="btn btn-secondary btn-sm" style="font-size:11px;padding:3px 8px;">测试</button>' +
        '<button onclick="deleteLuckMailConfig(' + idx + ')" class="btn btn-danger btn-sm" style="font-size:11px;padding:3px 8px;">删除</button>' +
      '</div>' +
    '</div>';
  });

  listDiv.innerHTML = html;
}

// 内联添加 LuckMail 配置
async function inlineAddLuckMail() {
  alert('inlineAddLuckMail 函数被调用了！');
  console.log('[LuckMail] inlineAddLuckMail called');
  var name = document.getElementById('luckmail-inline-name').value.trim();
  var token = document.getElementById('luckmail-inline-token').value.trim();
  var projectCode = document.getElementById('luckmail-inline-project').value.trim();
  var emailType = document.getElementById('luckmail-inline-emailtype').value.trim();
  var domain = document.getElementById('luckmail-inline-domain').value.trim();

  console.log('[LuckMail] token:', token, 'projectCode:', projectCode);

  if (!token) { showToast('请输入接口秘钥', 'error'); return; }
  if (!projectCode) { showToast('请输入项目代码', 'error'); return; }

  // 自动生成名称：包含邮箱类型和域名
  if (!name) {
    var nameParts = ['LuckMail'];
    if (emailType) nameParts.push(emailType);
    if (domain) nameParts.push(domain);
    if (nameParts.length === 1) nameParts.push(luckmailConfigs.length + 1);
    name = nameParts.join('-');
  }

  luckmailConfigs.push({
    name: name,
    token: token,
    projectCode: projectCode,
    emailType: emailType,
    domain: domain
  });

  var result = await window.go.main.App.SaveLuckMailConfigs(JSON.stringify(luckmailConfigs));
  if (result.error) {
    luckmailConfigs.pop();
    showToast(result.error, 'error');
    return;
  }

  // 只清空名称，保留默认值以便快速添加多个配置
  document.getElementById('luckmail-inline-name').value = '';
  // 恢复默认值
  document.getElementById('luckmail-inline-token').value = 'luck_a1209d58721677ccabdd828f0a7912da';
  document.getElementById('luckmail-inline-project').value = 'kiro';
  document.getElementById('luckmail-inline-emailtype').value = 'ms_imap';
  document.getElementById('luckmail-inline-domain').value = '';

  // 重置域名列表，恢复默认状态
  luckmailDomains = [];
  updateLuckMailDomainSelect();

  renderLuckMailInlineList();
  updateLuckMailSummary();
  showToast('LuckMail 配置已添加');

  // 同步更新注册页面的配置列表
  if (typeof loadLuckMailConfigToSelect === 'function') {
    loadLuckMailConfigToSelect();
  }
}

// 测试 LuckMail 连接
async function inlineTestLuckMail() {
  alert('inlineTestLuckMail 函数被调用了！');
  console.log('[LuckMail] inlineTestLuckMail called');
  var token = document.getElementById('luckmail-inline-token').value.trim();
  var projectCode = document.getElementById('luckmail-inline-project').value.trim();

  console.log('[LuckMail] token:', token, 'projectCode:', projectCode);

  if (!token) { showToast('请输入接口秘钥', 'error'); return; }
  if (!projectCode) { showToast('请输入项目代码', 'error'); return; }

  var btn = document.getElementById('luckmail-inline-test-btn');
  var statusEl = document.getElementById('luckmail-inline-status');
  btn.disabled = true;
  btn.textContent = '测试中...';
  statusEl.textContent = '';

  try {
    var config = JSON.stringify({
      name: 'test',
      token: token,
      projectCode: projectCode,
      emailType: document.getElementById('luckmail-inline-emailtype').value.trim(),
      domain: document.getElementById('luckmail-inline-domain').value.trim()
    });
    var result = await window.go.main.App.TestLuckMailConnection(config);
    if (result.error) {
      statusEl.innerHTML = '<span style="color:var(--danger);">' + escapeHtml(result.error) + '</span>';
    } else {
      statusEl.innerHTML = '<span style="color:var(--success);">连接成功, 余额: ¥' + (result.balance || 0).toFixed(2) + '</span>';

      // 更新域名列表
      if (result.domains && result.domains.length > 0) {
        luckmailDomains = result.domains;
        updateLuckMailDomainSelect();
        console.log('[LuckMail] 已更新域名列表:', luckmailDomains.length, '个域名');
      }
    }
  } catch (e) {
    statusEl.innerHTML = '<span style="color:var(--danger);">测试失败: ' + escapeHtml(e.message) + '</span>';
  }

  btn.disabled = false;
  btn.textContent = '测试连接';
}

// 更新域名下拉列表
function updateLuckMailDomainSelect() {
  var domainSelect = document.getElementById('luckmail-inline-domain');
  if (!domainSelect) return;

  var currentValue = domainSelect.value;
  var emailType = document.getElementById('luckmail-inline-emailtype').value.trim();
  var html = '<option value="">自动分配</option>';

  // API 域名优先；未获取到 API 域名时使用内置域名
  var sourceDomains = luckmailDomains.length > 0 ? luckmailDomains : defaultLuckMailDomains;

  // 根据选择的邮箱类型过滤域名
  var filteredDomains = sourceDomains;
  if (emailType) {
    filteredDomains = sourceDomains.filter(function(d) {
      return d.email_type === emailType;
    });
  }

  // 按域名排序（可选：也可以按数量排序）
  filteredDomains.sort(function(a, b) {
    // 如果有数量信息，按数量降序排序
    if (a.count && b.count) {
      return b.count - a.count;
    }
    // 否则按域名字母顺序
    return a.domain.localeCompare(b.domain);
  });

  filteredDomains.forEach(function(d) {
    var label = d.domain;
    // 如果有数量信息，显示数量
    if (d.count) {
      label += ' (' + d.count.toLocaleString() + ' 可用)';
    }
    html += '<option value="' + escapeHtml(d.domain) + '">' + escapeHtml(label) + '</option>';
  });

  domainSelect.innerHTML = html;

  // 尝试恢复之前的选择
  if (currentValue) {
    domainSelect.value = currentValue;
  }

  console.log('[LuckMail] 域名下拉列表已更新，显示', filteredDomains.length, '个域名');
}

// 获取域名列表
async function fetchLuckMailDomains() {
  var token = document.getElementById('luckmail-inline-token').value.trim();
  var projectCode = document.getElementById('luckmail-inline-project').value.trim();

  if (!token || !projectCode) return;

  try {
    var config = JSON.stringify({
      name: 'temp',
      token: token,
      projectCode: projectCode
    });
    var result = await window.go.main.App.GetLuckMailDomains(config);
    if (!result.error && result.domains) {
      luckmailDomains = result.domains;
      updateLuckMailDomainSelect();
      console.log('[LuckMail] 已获取域名列表:', luckmailDomains.length, '个域名');
    }
  } catch (e) {
    console.error('[LuckMail] 获取域名列表失败:', e);
  }
}

// 邮箱类型改变时更新域名列表
function onLuckMailEmailTypeChange() {
  updateLuckMailDomainSelect();
}

// 测试已保存的配置
async function testLuckMailConfig(idx) {
  var cfg = luckmailConfigs[idx];
  if (!cfg) return;

  try {
    var result = await window.go.main.App.TestLuckMailConnection(JSON.stringify(cfg));
    luckmailConfigStatus[cfg.name] = {
      tested: true,
      success: !result.error,
      balance: result.balance || 0
    };
    localStorage.setItem('luckmail-config-status', JSON.stringify(luckmailConfigStatus));
    renderLuckMailInlineList();

    if (result.error) {
      showToast('连接失败: ' + result.error, 'error');
    } else {
      showToast('连接成功, 余额: ¥' + (result.balance || 0).toFixed(2));
    }
  } catch (e) {
    showToast('测试失败: ' + e.message, 'error');
  }
}

// 删除配置
async function deleteLuckMailConfig(idx) {
  luckmailConfigs.splice(idx, 1);
  var result = await window.go.main.App.SaveLuckMailConfigs(JSON.stringify(luckmailConfigs));
  if (result.error) {
    showToast(result.error, 'error');
    await loadLuckMailConfigs();
    return;
  }
  renderLuckMailInlineList();
  updateLuckMailSummary();
  showToast('配置已删除');

  // 同步更新注册页面的配置列表
  if (typeof loadLuckMailConfigToSelect === 'function') {
    loadLuckMailConfigToSelect();
  }
}

// 页面加载时初始化
window.addEventListener('DOMContentLoaded', function() {
  console.log('[LuckMail] DOMContentLoaded 事件触发');
  console.log('[LuckMail] inlineAddLuckMail 函数存在:', typeof inlineAddLuckMail);
  console.log('[LuckMail] inlineTestLuckMail 函数存在:', typeof inlineTestLuckMail);
  loadLuckMailConfigs();
  updateLuckMailDomainSelect();
});

console.log('[LuckMail] luckmail.js 加载完成');

// 将函数暴露到全局作用域（确保 onclick 可以访问）
window.inlineAddLuckMail = inlineAddLuckMail;
window.inlineTestLuckMail = inlineTestLuckMail;
window.testLuckMailConfig = testLuckMailConfig;
window.deleteLuckMailConfig = deleteLuckMailConfig;
window.fetchLuckMailDomains = fetchLuckMailDomains;
window.onLuckMailEmailTypeChange = onLuckMailEmailTypeChange;

console.log('[LuckMail] 函数已暴露到 window 对象');
